### PR TITLE
[clojure][image] add draw-bounding-box interop

### PR DESCRIPTION
**Pull request by Chouffe**
## Description ##

* Added draw-bounding-box interop.
* Dependency: [https://github.com/apache/incubator-mxnet/pull/14474/](https://github.com/apache/incubator-mxnet/pull/14474/files)
* Resolves: [https://github.com/apache/incubator-mxnet/issues/14506](https://github.com/apache/incubator-mxnet/issues/14506)

## How it was tested

pug-cookie-4

```clojure
(-> image-path
    cv/imread
    (cvu/mat-to-buffered-image)
    (draw-bounding-box! [{:xmin 190 :xmax 850 :ymin 50 :ymax 450}
                         {:xmin 200 :xmax 350 :ymin 440 :ymax 530}]
                        {:stroke 2
                         :names ["pug" "cookie"]
                         :transparency 0.8
                         :font-size-mult 2.0})
    (cvu/buffered-image-to-mat)
    (cv/imwrite "pug-cookie-bounding-boxes.jpg"))
```

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by abhinavs95** [comment]
@mxnet-label-bot add [Clojure, pr-awaiting-review]

**Comment by gigasquid** [comment]
Thanks for this contribution. Will take a closer look in a bit 😸

File: contrib/clojure-package/src/org/apache/clojure_mxnet/image.clj [hunk 1]
```diff
@@ -199,3 +200,68 @@
   [input]
   (util/validate! ::to-image-ndarray input "Invalid input array")
   (Image/toImage input))
+
+(s/def ::buffered-image #(instance? BufferedImage %))
```
Comment (gigasquid) [inline contrib/clojure-package/src/org/apache/clojure_mxnet/image.clj]: Nice specs 😍

File: contrib/clojure-package/src/org/apache/clojure_mxnet/image.clj [hunk 2]
```diff
@@ -199,3 +200,68 @@
...
+(defn draw-bounding-box!
+  "Draw bounding boxes on `buffered-image` and Mutate the input image.
+  `buffered-image`: BufferedImage
+  `coordinates`: collection of {:xmin int :xmax int :ymin int :ymax int}
+  `font-size-mult`: positive float - Font size multiplier
+  `names`: collection of strings - List of names for the bounding boxes
+  `stroke`: positive integer - thickness of the bounding box
+  `transparency`: float in (0.0, 1.0) - Transparency of the bounding box
+  returns: Modified `buffered-image`
+  Ex:
```
Comment (gigasquid) [inline contrib/clojure-package/src/org/apache/clojure_mxnet/image.clj]: I really like the examples in the docstring

**Pull request merged**